src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.cs

```diff
@@ -569,8 +570,11 @@ private static string EntryFromPath(string entry, int offset, int length)
 
         private static Boolean IsDirEmpty(DirectoryInfo possiblyEmptyDir)
         {
-            foreach (FileSystemInfo fi in possiblyEmptyDir.EnumerateFileSystemInfos("*", SearchOption.AllDirectories))
-                return false;
+            using (IEnumerator<String> enumerator = Directory.EnumerateFileSystemEntries(possiblyEmptyDir.FullName).GetEnumerator())
+            {
+                if (enumerator.MoveNext())
+                    return false;
+            }
 
             return true;
```

Comment:
It occurs to me that now that this is using MoveNext instead of foreach, it could be simplified slightly:
```C#
private static bool IsDirEmpty(DirectoryInfo possiblyEmptyDir)
{
    using (IEnumerator<string> enumerator = Directory.EnumerateFileSystemEntries(possiblyEmptyDir.FullName).GetEnumerator())
        return !enumerator.MoveNext();
}
```